processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -727,13 +715,20 @@ public ResultRow apply(Object input)
             int dimPos = 0;
             while (dimsIter.hasNext() && results.hasNext()) {
               final DimensionSpec dimensionSpec = dimsIter.next();
+              final Object dimensionObject = results.next();
+              final Object dimensionObjectCasted;
 
               // Must convert generic Jackson-deserialized type into the proper type.
-              resultRow.set(
-                  dimensionStart + dimPos,
-                  DimensionHandlerUtils.convertObjectToType(results.next(), dimensionSpec.getOutputType())
-              );
-
+              if (dimensionSpec.getOutputType().is(ValueType.COMPLEX)) {
```

Comment:
i kind of wonder if we should special handle json (`COMPLEX<json>`) here since it is kind of a special complex type (that really should probably be a built-in standard type instead of complex...) because `convertValue` looks pretty expensive.

Reply:
> because convertValue looks pretty expensive.

That sounds reasonable.
Maybe as an alternative, what if we return `Object.class` instead of `StructuredValue.class` for `jsonStrategy.getClazz()`? I assume that convertValue should short-circuit in that case - converting it from an object to object. If that's the case, we'd get away with special handling (perhaps).